src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -3473,7 +3482,15 @@ s32 func_80836AB8(Player* this, s32 arg1) {
     return var;
 }
 
-void func_80836BEC(Player* this, PlayState* play) {
+/**
+ * Update Z Targeting if the Z button is pressed.
+ * This includes both sub-types of Z Targeting:
+ *     - Lock On: Player locks onto an actor and a reticle appears on the target.
```

Comment:
It's also worth noting that this function is still important for updating part of the targeting system where `Z button` is not pressed. For example, talking to actors, or forced target (think redead scream), so `Update Z Targeting if the Z button is pressed.` doesn't feel general enough

Reply:
Where is talking involved in this function?

Reply:
`cond = func_8083224C(play);` is checking if player has `ACTOR_FLAG_TALK` set, which then leads to player stateFlags being set when "talking" sets `this->unk_664` to the actor that is talking

Reply:
I see. This goes back to unk_664 being more than just `lockOnActor`.
I think it will probably hurt readability in like 90% of cases if the name isnt `lockOnActor`, but im still not really sure what to do :))

Reply:
I unnamed this function and removed the comment for now. It should at least wait until the Target PR is in, I suppose. Which we should work on trying to get ready soon :D